beta2-quickstart-master/frontend/src/hooks/useWallet.tsx

```diff
@@ -0,0 +1,29 @@
+import { WalletLocked } from 'fuels';
+import React, { useEffect, useMemo, useState } from 'react'
+import { useFuel } from './useFuel';
+import { useIsConnected } from './useIsConnected';
+
+const useWallet = () => {
+	const [account, setAccount] = useState<string>();
+	const [wallet, setWallet] = useState<WalletLocked | null>(null);
+	const isConnected = useIsConnected();
+	const [Fuel] = useFuel();
+
+	useEffect(() => {
+		async function getAccounts() {
+			const currentAccount = await Fuel.currentAccount();
```

Comment:
The current account can present issues if the user doesn't allow the existing account to connect. I suggest using `const [account] = await fuel.accounts()`.